tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1342,6 +1342,13 @@ This can be useful to create Callables to pass around without polluting the clas
     var lambda = func(x): print(x)
     lambda.call(42) # Prints "42"
 
+Unlike python, Lambda functions require an explicit `return`.
```

Comment:
Shouldn't reference python, GDScript isn't related to python and we shouldn't expect any knowledge like that

This should be expected as nowhere else do we have implicit return, so I don't think this is something to document

Reply:
I searched the document before deciding to mention python and found another part doing the same, so I thought it would be okay. 
It adds a valuable reference especially for people coming from python.
I would have never found out that they work this way if I hadn't ask in the chat about it.

At the very least an example with `return` could be provided.

Reply:
Then it's okay, didn't know there were, would need to be discussed still, as it's a very unusual assumption based on knowledge from another language

Reply:
```suggestion
Unlike Python, lambda functions require an explicit `return`.
```
Name, capitalize, which is done elsewhere, also lambda isn't a name

Reply:
To be honest, I think it is very natural to use knowledge from the closest language to gdscript that is out there.
No matter how often it is written that it's independent, people with according experience will at first assume it is like a python with less features (which it is to a large degree).

If, even after explicitly reading the lambda documentation, one is not made aware of this little difference, then the documentation is lacking imo.

Reply:
IMO that's not how to approach new situations, but I'm fine with adding this, but we can't necessarily point out every single way GDScript differs from Python, because it isn't Python, it only similar because it uses indentation, and the documentation clearly states it has no relation to it

Reply:
Mind you, even the [style guide](https://github.com/godotengine/godot-docs/blob/3107a52ce7a4a07e5dac5a8e74f1930f02b98a5c/tutorials/scripting/gdscript/gdscript_styleguide.rst#gdscript-style-guide) references python in the very first paragraph (saying it's close), as well as this document itself (if only to mention the indentation). 

It is put in peoples heads, whether you like it or not..

Reply:
As I said:
> GDScript is entirely independent from Python and is not based on it

But I'm fine with adding this, but I don't agree with the fundamental logic, but it's clearly stated to not be at all connected in that sense 🙂

Reply:
> but please fix the spelling issues

Yah, I was struggling with the desktop github client (don't often use it), it seems to be lacking the "amend" option, or its well hidden.

> As I said:
> GDScript is entirely independent from Python and is not based on it

I am not refuting this, but you have to also see that there clearly is a relation at it would be naive to not expect people to behave accordingly :) 
I am trying a pragmatic approach really. In my mind, it is rather likely that a new comer to godot (but not to programming) has previous python experience and will draw from it for this.
Maybe I am too biased from my own experience, but the smallest common denominator here would then be to have at least the example.

Let's see what others say or decide from here.

Reply:
Note though that this would require people to ignore warnings, so I'm not sure this is a critical thing to point out, it tells you you're doing something wrong if you do like that code

Reply:
as you mentioned warnings..

> Though I'm against it also because it causes errors and warnings, so it's clear you're doing something wrong

I did have a warning, true, but it was hard to understand because I thought it referenced the outer function

> winners.sort_custom(func (a, b): times[a]  < times[b])

so I thought it was an issue with `sort_custom` and left me more confused than helpful. 
My first assumption was that `sort_custom` was somehow wrongly tagged as "a function that returns something when in fact it doesn't" than the lambda being wrong..

And given that lambdas are usually embedded in other calls, it's probably not a rare scenario that way...